### ISSUE TITLE
Nesting markdown in block quotes gives recommonmark trouble

### ISSUE THREAD

**Issue by futuro** [clojurebridge-minneapolis/installfest#35]
`From Page_5_More_code_changes.md` in `docs/track1/`, line 518, the following snippet causes recommonmark to barf

``````
> #### conj
> There are many ways to work with collections of values in Clojure.  One commonly used function is `conj`. The name is short for "conjoin". This function takes a collection and one or more item(s) to add to the collection. It then returns a _new_ collection without modifying the original collection.
>
> ```clojure
> (conj [:one :two] :three)
> => [:one :two :three]
>
> (conj [:one :two :three] :four :five)
> => [:one :two :three :four :five]
> ```
``````

The error is

```
# Sphinx version: 1.4.8
# Python version: 3.5.2 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.8
# Last messages:
#   reading sources... [ 58%] setup_osx
#   reading sources... [ 62%] setup_ubuntu
#   reading sources... [ 65%] setup_win7
#   reading sources... [ 68%] setup_win8
#   reading sources... [ 72%] track1
#   reading sources... [ 75%] track1/Page_1_Intro_TOC
#   reading sources... [ 79%] track1/Page_2_Big_Picture
#   reading sources... [ 82%] track1/Page_3_Start_project
#   reading sources... [ 86%] track1/Page_4_Change_code
#   reading sources... [ 89%] track1/Page_5_More_code_changes
# Loaded extensions:
#   alabaster (0.7.9) from /home/user/.virt-envs/installfest/lib/python3.5/site-packages/alabaster/__init__.py
Traceback (most recent call last):
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/sphinx/cmdline.py", line 244, in main
    app.build(opts.force_all, filenames)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/sphinx/application.py", line 297, in build
    self.builder.build_update()
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/sphinx/builders/__init__.py", line 265, in build
    self.doctreedir, self.app))
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/sphinx/environment.py", line 569, in update
    self._read_serial(docnames, app)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/sphinx/environment.py", line 589, in _read_serial
    self.read_doc(docname, app)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/sphinx/environment.py", line 742, in read_doc
    pub.publish()
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/sphinx/io.py", line 49, in read
    self.parse()
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/recommonmark/parser.py", line 87, in parse
    self.convert_block(ast)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/recommonmark/parser.py", line 49, in convert_block
    self.convert_blocks(block.children)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/recommonmark/parser.py", line 45, in convert_blocks
    self.convert_block(block)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/recommonmark/parser.py", line 55, in convert_block
    self.blockquote(block)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/recommonmark/parser.py", line 139, in blockquote
    self.convert_blocks(block.children)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/recommonmark/parser.py", line 45, in convert_blocks
    self.convert_block(block)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/recommonmark/parser.py", line 51, in convert_block
    self.section(block)
  File "/home/user/.virt-envs/installfest/lib/python3.5/site-packages/recommonmark/parser.py", line 110, in section
    self.current_node.document.note_implicit_target(new_section, new_section)
AttributeError: 'NoneType' object has no attribute 'note_implicit_target'
```

I'm looking into whether we can maintain the current markdown and get this fixed upstream.

**Comment by tmarble** [comment]
Working with upstream is a great goal... In the meantime do we have a workaround? There must be a way to have code blocks in RTD -- I'm sure i've seen it in Python docs for example.

**Comment by futuro** [comment]
So the issue is that we're nesting markdown inside of block quotes. For example, the following line causes problems.

```
> ### Header
```

You can see how this renders in github if you go to https://github.com/clojurebridge-minneapolis/installfest/blob/integrate-track1/docs/track1/Page_5_More_code_changes.md#conj

The grayed out effect of that section is lost if you can't nest markdown inside of block quotes. If we don't care about that grayed out effect then we could simply remove the block quotes for nested objects.

**Comment by tmarble** [comment]
Seems like an easy solution is hoist the header outside of the quote block?

```
To add messages to `chat-messages`, we will need to introduce two more functions: `conj` and `swap!`.

#### conj
> There are many ways to work with collections of values in Clojure.  ...
```

**Comment by futuro** [comment]
I hadn't even thought to test which particular nested markdown causes issues, but it turns out it's just headers in block quotes causes recommonmark issues. So yeah, we could hoist those outside of the block quotes for the time being.

**Comment by futuro** [comment]
Filed rtfd/recommonmark#57

**Comment by futuro** [comment]
We can have code blocks, but attempting to put markdown inside of a block
quote seems to be causing issues. So if we remove the block quote markers
from the nested markdown sections it compiles fine. I'll try to include an
example of what I mean tonight.

**Comment by futuro** [comment]
I've resolved this through PR #39 

If upstream releases a new version we should change the requirements back to a particular version.
